opencompass/tasks/notdiamond_eval.py

```diff
@@ -332,8 +338,10 @@ def _log_eval_success_to_wandb(self, result: dict, metric: str):
         details = result["details"]
```

Comment:
Can you check if `result["details"]` contains the same information as the dataset json file samples? If so then this looks good.

Reply:
Ah, so that's a great catch - it actually doesn't. At the moment we do not link our training dataset samples back to the originating eval dataset samples. So it would not be straightforward from this point (within the eval task) to link back to the original eval dataset sample's ID.

I think we might have to write failed eval dataset samples during a inference task, when we can actually access the original sample. What do you think?

Reply:
I actually think you can. Because you have the `sample_id` and you know the dataset `abbr`. You can also access `db_url` from `self.dataset_cfg`. That should be enough to identify the dataset json file and extract the desired sample using `sample_id`

Reply:
That's fair, could definitely do that.